[Bug] [dlink-gateway] 启动k8s session集群时报空指针异常

### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar issues.


### What happened

配置k8s session集群之后，启动时报如下错误：
```
[dlink] 2023-03-13 16:20:38.100  ERROR 13603 --- [nio-8888-exec-7] com.dlink.utils.LogUtil: org.apache.flink.client.deployment.ClusterDeploymentException: Could not create Kubernetes cluster "test-k8s-cluster".
	at org.apache.flink.kubernetes.KubernetesClusterDescriptor.deployClusterInternal(KubernetesClusterDescriptor.java:291)
	at org.apache.flink.kubernetes.KubernetesClusterDescriptor.deploySessionCluster(KubernetesClusterDescriptor.java:160)
	at com.dlink.gateway.kubernetes.KubernetesSessionGateway.deployCluster(KubernetesSessionGateway.java:90)
	at com.dlink.job.JobManager.deploySessionCluster(JobManager.java:738)
	at com.dlink.service.impl.ClusterServiceImpl.deploySessionCluster(ClusterServiceImpl.java:178)
	at com.dlink.service.impl.ClusterServiceImpl$$FastClassBySpringCGLIB$$d1f6e853.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy.invokeMethod(CglibAopProxy.java:386)
	at org.springframework.aop.framework.CglibAopProxy.access$000(CglibAopProxy.java:85)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:704)
	at com.dlink.service.impl.ClusterServiceImpl$$EnhancerBySpringCGLIB$$559aa8ff.deploySessionCluster(<generated>)
	at com.dlink.controller.ClusterController.deploySessionCluster(ClusterController.java:197)
	at com.dlink.controller.ClusterController$$FastClassBySpringCGLIB$$e7b26861.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.adapter.AfterReturningAdviceInterceptor.invoke(AfterReturningAdviceInterceptor.java:57)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.dlink.controller.ClusterController$$EnhancerBySpringCGLIB$$8ff7a4d1.deploySessionCluster(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1071)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:964)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:670)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:779)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at com.alibaba.druid.support.http.WebStatFilter.doFilter(WebStatFilter.java:124)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:893)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1789)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at org.apache.flink.configuration.ConfigurationUtils.convertToString(ConfigurationUtils.java:388)
	at org.apache.flink.configuration.ConfigurationUtils.lambda$convertToString$11(ConfigurationUtils.java:396)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Collections$2.tryAdvance(Collections.java:4719)
	at java.util.Collections$2.forEachRemaining(Collections.java:4727)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at org.apache.flink.configuration.ConfigurationUtils.convertToString(ConfigurationUtils.java:397)
	at org.apache.flink.configuration.Configuration.toMap(Configuration.java:741)
	at org.apache.flink.configuration.ConfigurationUtils.getPrefixedKeyValuePairs(ConfigurationUtils.java:268)
	at org.apache.flink.kubernetes.kubeclient.parameters.KubernetesJobManagerParameters.getEnvironments(KubernetesJobManagerParameters.java:84)
	at org.apache.flink.kubernetes.kubeclient.decorators.InitJobManagerDecorator.getCustomizedEnvs(InitJobManagerDecorator.java:174)
	at org.apache.flink.kubernetes.kubeclient.decorators.InitJobManagerDecorator.decorateMainContainer(InitJobManagerDecorator.java:145)
	at org.apache.flink.kubernetes.kubeclient.decorators.InitJobManagerDecorator.decorateFlinkPod(InitJobManagerDecorator.java:100)
	at org.apache.flink.kubernetes.kubeclient.factory.KubernetesJobManagerFactory.buildKubernetesJobManagerSpecification(KubernetesJobManagerFactory.java:79)
	at org.apache.flink.kubernetes.KubernetesClusterDescriptor.deployClusterInternal(KubernetesClusterDescriptor.java:273)
```

### What you expected to happen

能够正常启动session集群

### How to reproduce

类：KubernetesSessionGateway
方法：deployCluster
修改位置：
```
 configuration.set(PipelineOptions.JARS, Collections.singletonList(appConfig.getUserJarPath()));
```
添加对appConfig.getuserJarPath()返回值进行非空判断
```
if (appConfig.getUserJarPath() != null) {
    configuration.set(PipelineOptions.JARS, Collections.singletonList(appConfig.getUserJarPath()));
}
```

### Anything else

_No response_

### Version

0.7.2

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dlink-gateway/src/main/java/com/dlink/gateway/kubernetes/KubernetesSessionGateway.java
+++ b/dlink-gateway/src/main/java/com/dlink/gateway/kubernetes/KubernetesSessionGateway.java
@@ -61,13 +61,13 @@ public class KubernetesSessionGateway extends KubernetesGateway {
         AppConfig appConfig = config.getAppConfig();
         String flinkConfigPath = config.getClusterConfig().getFlinkConfigPath();
         Configuration loadConfiguration = GlobalConfiguration.loadConfiguration(flinkConfigPath);
-        if (loadConfiguration != null) {
-            loadConfiguration.addAll(configuration);
-            configuration = loadConfiguration;
-        }
+        loadConfiguration.addAll(configuration);
+        configuration = loadConfiguration;
         configuration.set(DeploymentOptionsInternal.CONF_DIR, flinkConfigPath);
 
-        configuration.set(PipelineOptions.JARS, Collections.singletonList(appConfig.getUserJarPath()));
+        if (Asserts.isNotNullString(appConfig.getUserJarPath())) {
+            configuration.set(PipelineOptions.JARS, Collections.singletonList(appConfig.getUserJarPath()));
+        }
 
         KubernetesClusterDescriptor kubernetesClusterDescriptor = new KubernetesClusterDescriptor(configuration,
                 client);

